### DISCUSSION TITLE
Notification channel alerts no longer vibrate

### DISCUSSION THREAD

**Discussion by richard-dennehy** [NightscoutFoundation/xDrip#3208] [Bug reports]
**Xdrip version:** 2123112109
**Phone:** Google Pixel 6a
**Android version:** 14 (1 November 2023 security update)

After installing the latest xdrip version, my high/low alert notifications have stopped vibrating. Disabling "notification channels" fixes this, but as far as I'm aware, I've always had that enabled, and never had an issue before.

I've tried using the Android notification settings to enable vibrations on notifications in the "Glucose Alerts Settings" channel, but this just enables the default Android notification vibration, which is very easily missed.

**Comment by Navid200** [comment]
>> but this just enables the default Android notification vibration  

It's not clear what you expect.

If you are happy with the options xDrip gives you for customizing your alerts, why are you enabling notifications channels?

If you like to use notifications channels, so that you can further customize using Android, why are you surprised that Android is taking over and you will need to do your customization using Android notification channel settings.

It's one or the other.  Have you ever had a third scheme that is now disappeared?

**Reply by richard-dennehy** [reply]
To be clear: this isn't a setting I've made any changes to recently - my alerts stopped working on Thursday after installing the latest xDrip version, and after spending an hour or so looking through Android and xDrip settings, I noticed turning off Notification Channels fixes it.

My concern is that this could cause other people to miss important alerts

**Comment by nsghub** [comment]
I can confirm this, on the same device and the same Android version. No vibrations or audi notifications if "Notification Channels" is enabled. After disabling this option, everything is back to normal.
![Screenshot_20231126-102245](https://github.com/NightscoutFoundation/xDrip/assets/39347827/ea88012b-3d83-4229-8e98-3a6f0b377b58)

**Comment by Navid200** [comment]
Thanks for reporting this.

If you install xDrip on Android 14, you will get an Android notification that the app you are installing is meant for an older version of Android.
That is an incorrect statement.  xDrip is tested and verified to be operational with Android 14.  But is forced on us by Google.

We update xDrip to be operational with the latest Android.  But, we would also like xDrip to be usable with older Android releases because not everyone all over the world can afford to get the latest and the greatest.

The time has come to gradually make some updates so that we can retire Android 6.  In the process, you may experience some changes.  I suspect this is one of those.
So, I believe you when you say that you made no change other than updating xDrip.

But, my question for you is why is this a problem?
You want the alerts to enable the vibration.  And not using Google's Notification channels, will let you have the vibration work.  So, why do you want to enable notification channels?

**Reply by richard-dennehy** [reply]
I suppose it's not a problem for me any more, since I've found a fix, but my expectation is that when I update xDrip, everything will behave the same as before, within reason.

I thought I'd raise this as a potentially unexpected change in behaviour, especially since it's affecting alarms. If you think this is working as expected, then I suppose there's no issue here.

**Reply by Navid200** [reply]
I'm not the lead developer.
What I have stated is my personal opinion.

We have never recommended using notification channels.  The behavior is not predictable on all phone brands and models.

With reference to what I consider acceptable, I, again my personal opinion, would consider a low alert not triggering after an update unacceptable.
The vibration component of an alert sounding differently is an acceptable inconvenience in my opinion.

If we say, we will never update xDrip unless the behavior will not change in any shape or form, xDrip will become outdated soon. 
I am not saying what you are experiencing is not an inconvenience.  It is.  I wish we could avoid it.  And for all I know, the lead developer may spend a lot of time trying to find a workaround for this.
I prefer he focuses on what is really critical and a show stopper.  
But, whether this is a show stopper or not is something he has to decide.  That's what makes  him the lead.  

Again, thanks for reporting this and thanks for your patience.

**Comment by nsghub** [comment]
For me this behaviour of xdrip was totally unexpected. The alarms were triggered visibly only but not audibly nor by vibration. I had this option enabled for many many xdrip versions without any issues.

I must admit that I do not know what @richard-dennehy means with "Android notification vibration".

**Reply by richard-dennehy** [reply]
You can configure the notification channel here - I was hoping that enabling vibration here would help, but this just enables the standard vibration which is very easily missed

![Screenshot_20231126-145229](https://github.com/NightscoutFoundation/xDrip/assets/14271785/0807a2c8-1e0b-4d84-8950-5b4a7d8951d1)

**Reply by nsghub** [reply]
Ah ok. I had/have this enabled but it did not make a difference for me.

**Reply by Navid200** [reply]
@nsghub 
I'm sorry about the trouble.
Please disable notification channels for now and use xDrip to customize your alerts.
Let me know if you run into trouble and I will help you.

In the future, my plan is to enhance the alert controls.
For example, I plan to add the ability to customize the vibration patterns in xDrip itself.
There is a lot I intend to do.  But, it takes a little bit of time.

**Reply by nsghub** [reply]
@Navid200 
I am fine now. Disabling this Android 8+ option fixed it. But it is weird that it happend at all. By the way, the alerts were quiet on my Android Wear watch, too.

**Comment by jamorham** [comment]
This is really annoying and is likely due to the updated support library version which was recently introduced on the nightly channel. Notification channels is one aspect where they have changed what we can access in the framework api. I attempted to work around it, but obviously insufficiently based on these problem reports.

**Comment by Navid200** [comment]
I am working on a new feature that will let you choose one of different vibration patterns for each one of your alerts.  
I expect to submit it in a week or two.  If it is approved, you will be able to customize the vibration, which you will not be able to do with Notification Channels.

**Comment by jamorham** [comment]
In the latest nightly there is a new workaround for this, please let me know if it resolves the problem and you're able to use notification channels again effectively.

**Reply by richard-dennehy** [reply]
Yeah, that works as it did before. Thanks 👍

**Reply by jamorham** [reply]
Ok, so just to confirm you have re-enabled the notification channels on this version?

**Reply by richard-dennehy** [reply]
Yeah, I re-enabled notification channels and triggered a test alert, and it vibrated as it used to

**Comment by nsghub** [comment]
I tested the latest Nightly with the Android 8+ Notification Channels feature and the alerts play a sound and vibrate again.

05.12.2023 18:34:17 JamOrHam ***@***.***>:
